langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/gemini/GeminiAiChatLanguageModelIT.java

```diff
@@ -0,0 +1,597 @@
+import static dev.langchain4j.model.gemini.GeminiHarmBlockThreshold.BLOCK_LOW_AND_ABOVE;
+import static dev.langchain4j.model.gemini.GeminiHarmCategory.HARM_CATEGORY_HARASSMENT;
+import static dev.langchain4j.model.gemini.GeminiHarmCategory.HARM_CATEGORY_HATE_SPEECH;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(RetryExtension.class)
+public class GeminiAiChatLanguageModelIT {
+
+    private static final String GEMINI_AI_KEY = System.getenv("GEMINI_AI_KEY");
+
+    private static final String CAT_IMAGE_URL = "https://upload.wikimedia.org/wikipedia/commons/e/e9/Felis_silvestris_silvestris_small_gradual_decrease_of_quality.png";
+    private static final String MD_FILE_URL = "https://raw.githubusercontent.com/langchain4j/langchain4j/main/docs/docs/intro.md";
+
+    @Test
+    void should_answer_simple_question() {
+        // given
+        GeminiAiChatLanguageModel gemini = GeminiAiChatLanguageModel.builder()
+            .apiKey(GEMINI_AI_KEY)
+            .modelName("gemini-1.5-flash")
+            .build();
+
+        // when
+        ChatResponse response = gemini.chat(ChatRequest.builder()
+            .messages(UserMessage.from("What is a large language model?"))
+            .build());
+
+        // then
+        String text = response.aiMessage().text();
+        assertThat(text).containsIgnoringCase("artificial intelligence");
+        assertThat(text).containsIgnoringCase("neural network");
+
+        assertThat(response.finishReason()).isEqualTo(FinishReasonMapper.fromGFinishReasonToFinishReason(GeminiFinishReason.STOP));
+    }
+
+    @Test
+    void should_configure_generation() {
+        // given
+        GeminiAiChatLanguageModel gemini = GeminiAiChatLanguageModel.builder()
+            .apiKey(GEMINI_AI_KEY)
+            .modelName("gemini-1.5-flash")
+            .temperature(2.0)
+            .topP(0.5)
+            .topK(10)
+            .maxOutputTokens(10)
+            .build();
+
+        // when
+        String response = gemini.generate(
+            "How much is 3+4? Reply with just the answer");
+
+        // then
+        assertThat(response.trim()).isEqualTo("7");
+    }
+
+    @Test
+    void should_answer_in_json() {
+        // given
+        GeminiAiChatLanguageModel gemini = GeminiAiChatLanguageModel.builder()
+            .apiKey(GEMINI_AI_KEY)
+            .modelName("gemini-1.5-pro")
+            .responseMimeType("application/json")
+            .build();
+
+        // when
+        Response<AiMessage> response = gemini.generate(
+            UserMessage.from("What is the firstname of the John Doe?\n" +
+                "Reply in JSON following with the following format: {\"firstname\": string}"));
+
+        // then
+        String jsonText = response.content().text();
+        assertThat(jsonText).contains("\"firstname\"");
+        assertThat(jsonText).contains("\"John\"");
```

Comment:
```suggestion
```

Reply:
Why removing this line? To weaken the assertions a bit?

Reply:
please check suggestion for the previous line, "John" is checked there